### PR TITLE
Add shared-lib-not-executable check for libraries without exec bit se…

### PR DESCRIPTION
**Pull request by marxin**
…t (#94).

### CODE REVIEW

**Comment by marxin** [comment]
Related issue #7.

**Comment by marxin** [comment]
Patch has been updated to report only for shared libs that have `INTERP` header set.

File: rpmlint/checks/BinariesCheck.py [hunk 1]
```diff
@@ -261,6 +262,12 @@ def _check_forbidden_functions(self, pkg, pkgfile_path, path):
         for fn in forbidden_functions_filtered:
             self.output.add_info('W', pkg, fn, path, forbidden_functions[fn]['f_name'])
 
+    def _check_executable_shlib(self, pkg, pkgfile_path, path):
+        if not self.is_exec and self.readelf_parser.is_shlib:
+            interp = [h for h in self.readelf_parser.program_header_info.headers if h.name == 'INTERP']
+            if len(interp):
```
Comment (scarabeusiv) [inline rpmlint/checks/BinariesCheck.py]: `if interp:` is good enough here, empty lists are False so you don't need to calculate the length of it.

File: rpmlint/checks/BinariesCheck.py [hunk 2]
```diff
@@ -261,6 +262,12 @@ def _check_forbidden_functions(self, pkg, pkgfile_path, path):
...
+                self.output.add_info('W', pkg, 'shared-lib-not-executable', path)
```
Comment (scarabeusiv) [inline rpmlint/checks/BinariesCheck.py]: Also as it is really limited on which libs need this and they must be executable I would say this should be error.

**Pull request merged**